ui/src/views/PhotoView.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="row">
+    <div class="col-md-10 border border-3 rounded-3">
+      <img class="img-fluid" :src="src" />
+    </div>
+    <div class="col-md-2 border border-3 rounded-3">
+      <Praise :photo-id="id" :api="api" />
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-md-10 border border-3 rounded-3">
+      <Description :api="api" :photo="photo" />
+    </div>
+    <div class="col-md-2 border border-3 rounded-3">
+      <PhotoActions :api="api" :photo="photo" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Options, Vue } from "vue-class-component";
+import Praise from "@/components/Praise.vue";
+import PhotoActions from "@/components/PhotoActions.vue";
+import Description from "@/components/Description.vue";
+import { ApiThing, Photo, User } from "@/ApiThing";
+
+@Options({
+  components: { Praise, PhotoActions, Description },
+  props: { id: "", api: ApiThing }
+})
+export default class PhotoView extends Vue {
+  id = "";
+  api: ApiThing | null = null;
+  photo: Photo | null = null;
+  user: User | null = null;
+
+  mounted() {
+    this.api
+      ?.getPhoto(this.id)
```

Comment:
Pls use the `aync/await` syntax for better code readablity: typescriptlang.org/docs/handbook/release-notes/typescript-1-7.html

Reply:
Same for other views

Reply:
Not sure I understand what you mean.
So replace the use of promises with await?
The link you posted just leads me to typescript.org

Reply:
Yes, those promises look like spaghetti code, so please use the `async/await` syntax instead.